### PR TITLE
tensorflow eager tutorial

### PR DESCRIPTION
**Pull request by pesser**
this pr replicates the pytorch tutorial using tensorflow eager execution. in addition it adds the following features:
- include standard datasets to quickly get started: `edflow.datasets.mnist.MNIST` and `edflow.datasets.fashionmnist.FashionMNIST` for now. they automatically download to `~/.eddata_cache` on first instantiation and by default return train split in train mode and test split in test mode, but the split can be specified manually, too.
- `edflow.hooks.checkpoint_hooks.lambda_checkpoint_hook.LambdaCheckpointHook`: specify path to checkpoint directory, functions to get and set global step and functions to save and restore a model and it can be used as a functor to load the model and set the global step (e.g. for use in initialize) as well as a hook to write checkpoints each epoch or at a specified frequency (+ at exceptions)

### CODE REVIEW

**Review by jhaux — APPROVED** [review]
Nice work! Will check out the Lambda CheckpointHook sometime soon!
